plugins/discovery-ec2/src/test/java/org/elasticsearch/discovery/ec2/AmazonEC2Mock.java

```diff
@@ -520,6 +553,11 @@ public void deleteKeyPair(DeleteKeyPairRequest deleteKeyPairRequest) throws Amaz
     }
 
     @Override
+    public DeleteNatGatewayResult deleteNatGateway(DeleteNatGatewayRequest deleteNatGatewayRequest) {
+        return null;
```

Comment:
Actually, I should here also send `throw new UnsupportedOperationException("Not supported in mock"); `

I'll update the PR